onnxruntime/core/providers/qnn/qnn_provider_factory.cc

```diff
@@ -219,9 +219,11 @@ struct QnnEpFactory : OrtEpFactory {
         OrtKeyValuePairs* ep_options = nullptr;
         factory->ort_api.CreateKeyValuePairs(&ep_options);
         factory->ort_api.AddKeyValuePair(ep_options, "backend_path", factory->qnn_backend_path.c_str());
-        ORT_API_RETURN_IF_ERROR(
-            factory->ort_api.GetEpApi()->CreateEpDevice(factory, &device, nullptr, ep_options,
-                                                        &ep_devices[num_ep_devices++]));
+        OrtStatus* status = factory->ort_api.GetEpApi()->CreateEpDevice(factory, &device, nullptr, ep_options,
+                                                                        &ep_devices[num_ep_devices++]);
+
+        factory->ort_api.ReleaseKeyValuePairs(ep_options);
```

Comment:
The alternative is to use the C++ API but that would require a larger code change (handle exceptions, initialize C++ global api object).

Reply:
Not really, you could capture status with Ort::Status and the same for keyvalue pairs and then decide what to do next.

Reply:
I'd prefer to defer that change. QNN EP is built as both a static and shared library, which impacts how the C++ api object should be initialized. Also, while one could mix use of the C API with C++ api to avoid exceptions, I'd prefer to let QC handle that to keep this change small.